app/api/v1/endpoints/auth.py

```
@@ -266,13 +266,20 @@ async def login_for_access_token(
 
         # Authenticate user (OAuth2 uses 'username' field for email)
-        user = user_service.authenticate_user(form_data.username, form_data.password)
-        if not user:
+        try:
+            user = user_service.authenticate_user(form_data.username, form_data.password)
+        except InvalidCredentialsError:
             raise HTTPException(
                 status_code=401,
                 detail="Incorrect email or password",
                 headers={"WWW-Authenticate": "Bearer"},
             )
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)